src/python/pants/backend/native/register.py

```diff
@@ -0,0 +1,24 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+from pants.backend.native.subsystems.native_toolchain import NativeToolchain
+from pants.engine.rules import RootRule, SingletonRule
+from pants.util.objects import datatype
+from pants.util.osutil import get_normalized_os_name
+
+
+class Platform(datatype('Platform', ['normalized_os_name'])):
```

Comment:
I think this should have a docstring, if you've got a good idea for one handy.

Reply:
I do -- this datatype was ripped from my real dev branch [here](https://github.com/cosmicexplorer/pants/tree/add-lld-to-native-toolchain) which has a ton more changes (and Platform isn't defined in register.py, for example), so I was a little hasty. I want to leave the docstring off for now because I'm going to move this class out of register.py *for sure* immediately after this pr gets merged (and keeping the docstring out makes it less legitimate).

Does that make sense? I can totally add the docstring too with great ease, just wanted to get your thoughts on my thoughts.

Reply:
That works for me.